src/connectivity.jl

```diff
@@ -118,72 +118,105 @@ Return `true` if the directed graph `g` is connected.
 function is_weakly_connected end
 @traitfn is_weakly_connected(g::::IsDirected) = length(weakly_connected_components(g)) == 1
 
-# Adapated from Graphs.jl
-mutable struct TarjanVisitor{T<:Integer} <: AbstractGraphVisitor
+
+mutable struct TarjanState{T<:Integer}
     stack::Vector{T}
-    onstack::BitVector
+    onstack::Vector{Bool}
     lowlink::Vector{T}
     index::Vector{T}
+    parents::Vector{T}
     components::Vector{Vector{T}}
 end
 
-TarjanVisitor(n::T) where T<:Integer = TarjanVisitor(
-    Vector{T}(),
-    falses(n),
+TarjanState(n::T) where T<:Integer = TarjanState(
     Vector{T}(),
+    zeros(Bool, n),
+    zeros(T, n),
+    zeros(T, n),
     zeros(T, n),
     Vector{Vector{T}}()
 )
```

Comment:
What if we got rid of `TarjanState` completely and moved the struct components inside the function itself?